samples/matter/common/src/bridge/ble_connectivity_manager.cpp

```diff
@@ -411,10 +413,8 @@ CHIP_ERROR BLEConnectivityManager::RemoveBLEProvider(bt_addr_le_t address)
 		return CHIP_ERROR_INTERNAL;
 	}
 
-	/* Release the connection immediately in case of disconnection failed, as callback will not be called. */
-	if (bt_conn_disconnect(provider->GetBLEBridgedDevice().mConn, BT_HCI_ERR_REMOTE_USER_TERM_CONN) != 0) {
-		bt_conn_unref(provider->GetBLEBridgedDevice().mConn);
-	}
+	bt_conn_disconnect(provider->GetBLEBridgedDevice().mConn, BT_HCI_ERR_REMOTE_USER_TERM_CONN);
```

Comment:
What will happen if this function returns an Error? Shouldn't we return at least an error code?

Reply:
The `RemoveBLEProvider` method is called in one of two scenarios, adding a bridged device corresponding to this provider failed, so we have to delete it or the provider destructor was called, so it will be dead soon anyway.
Returning an error would suggest that the removal process failed, which is not true. I don't know how we should handle this error, as if disconnect failed it mean we were not able to close the connection gently, and we will do that brutally in the next step and unref the connection.

Reply:
ok, I see, so let's leave it as it is :)